testsuite/MDAnalysisTests/coordinates/base.py

```diff
@@ -512,6 +511,22 @@ def test_iter_as_aux_lowf(self, ref, reader):
                                          ref.iter_ts(ref.aux_lowf_frames_with_steps[i]),
                                          decimal=ref.prec)
 
+    def test_iter_rewinds(self, reader):
+        for ts_indices in reader[[0, 1, 2]]:
+            pass
+        for ts_sliced in reader[:3]:
+            pass
+        assert_equal(ts_indices.frame, ts_sliced.frame,
+                     err_msg="FrameIteratorIndices and Sliced do not match")
+        assert_equal(ts_indices.frame, 0,
+                     err_msg="FrameIteratorIndices did not rewind")
+        assert_equal(ts_sliced.frame, 0,
+                     err_msg="FrameIteratorSliced did not rewind")
+        for ts_all in reader:
+            pass
+        assert_equal(ts_all.frame, 0,
+                     err_msg="FrameIteratorAll did not rewind")
```

Comment:
These are three different tests. Split it into three different methods to keep things simple. You might be able to write one test and then [parametrize](https://docs.pytest.org/en/6.2.x/parametrize.html) with three accessor functions (and different [test ids](https://docs.pytest.org/en/6.2.x/example/parametrize.html#paramexamples) so that it looks nice in the output)
```python
@pytest.mark.parametrize("accessor", [
          lambda traj: traj[[0, 1, 2]], 
          lambda traj: traj[:3],
          lambda traj: traj],
          ids=["indexed", "sliced", "all"])
def test_iter_rewinds(self, reader, accessor):
   for ts in accessor(reader):
        ...
   ...
```

Reply:
Hi @orbeckst,

@hmacdope previously suggested "I would love to see a test in testsuite/MDAnalysisTests/coordinates/base.py called something like test_iteration_rewinds where you compare where the trajectory ends up from the different types of iteration."

which is what I was trying to do with the assert referenced below. I was wondering if you think I explicitly should compare the that they end up at the same frame or if its better to just test that all the iterators end up rewinding to 0?
```
assert_equal(ts_indices.frame, ts_sliced.frame,
                     err_msg="FrameIteratorIndices and Sliced do not match")
```

Reply:
I check the most specific thing, namely that they rewind to zero.